nix/magic-folder.nix

```diff
@@ -17,7 +17,20 @@ buildPythonPackage rec {
   ];
 
   postPatch = ''
-    PATH="$PATH:${git}/bin" ${python}/bin/python setup.py update_version
+    # Generate _version.py ourselves since we can't rely on the Python code
+    # extracting the information from the .git directory that is likely not
+    # available.
+    cat > src/magic_folder/_version.py <<EOF
+
+# This _version.py is generated from metadata by nix/magic-folder.nix.
+
+__pkgname__ = "magic-folder"
+real_version = "${version}"
+full_version = "${version}"
+branch = "main"
+verstr = "${version}"
+__version__ = verstr
```

Comment:
I guess nothing currently cares, but I'm slightly leary of having a non-PEP 440 version here.

Thinking about it a little more, I wonder if we can get away with using `"unknown"` for the version if we don't have one (which is what tahoe does); we could set it in `magic_folder/__init__.py` if we can't import from `._version`? Though that would mean we won't notice if we deploy without a proper version. That still would be a non-PEP 440 version, but not version-like, which I think is better; it also feels more honest that a random date that doesn't get updated.

We could address the later issue for gridsync by adding some sort of integration test in that asserts the version isn't unknown.

Reply:
I can replace the "-" with "." to make it PEP 440.  I could also mark this as a "dev" or "pre" version or some such to make it more clear it's not an actual final/release version.

Reply:
I'm definitely wary about using a version with a different versioning scheme than the project uses[1]. I know that apple used (uses?) a different version scheme for clang than upstream, which has caused confusion. That is why I'd prefer `"unknown"` than something date based, even if it changed to be PEP440-compliant (without deciding to go with calver for magic-folder).

[1] I'd be fine switching to something calver based, but if we do that, we should be explicit about that change.

Reply:
The version number is now PEP 440-based.  It still adheres to the semver-ish scheme that the other packaging was using.  I don't mind switching to calver or something but that seems like a separate change.